service/src/test/java/bio/terra/profile/service/profile/ProfileServiceUnitTest.java

```diff
@@ -236,6 +241,38 @@ void getProfileWithEnterpriseOrganization() throws InterruptedException {
     assertFalse(nonEnterpriseResult.organization().get().isEnterprise());
   }
 
+  @Test
+  void getProfileWithOrganizationLimits() throws InterruptedException {
+    var organizationSubscription = UUID.randomUUID();
+    Map<String, String> limitMap = Map.of("vcpus", "4");
+    when(limitsConfiguration.subscriptions())
+        .thenReturn(Map.of(organizationSubscription, limitMap));
+    var limitedProfile =
+        ProfileFixtures.createAzureBillingProfile(
+            UUID.randomUUID(), organizationSubscription, "limitedMRG");
+    var nonLimitedProfile =
+        ProfileFixtures.createAzureBillingProfile(
+            UUID.randomUUID(), UUID.randomUUID(), "nonLimitedMRG");
+
+    when(profileDao.getBillingProfileById(limitedProfile.id())).thenReturn(limitedProfile);
+    when(profileDao.getBillingProfileById(nonLimitedProfile.id())).thenReturn(nonLimitedProfile);
+    when(samService.hasActions(eq(user), eq(SamResourceType.PROFILE), any())).thenReturn(true);
+    when(tpsApiDispatch.getOrCreatePao(any(), any(), any())).thenReturn(new TpsPaoGetResult());
+
+    var limitedResult = profileService.getProfile(limitedProfile.id(), user);
+    var nonLimitedResult = profileService.getProfile(nonLimitedProfile.id(), user);
+
+    Object limitsObject = limitedResult.organization().get().getLimits();
+    assertTrue(limitsObject instanceof Map);
```

Comment:
This assertion seems redundant, how would getLimits come back as anything other than a Map (or null)? You might be able to condense L265-268 by making `limitsObject` a var and leaving the assertion on L268.